lib/rubocop/cop/rails/squished_sql_heredocs.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      #
+      # Checks multi- and singleline SQL heredocs to use `.squish`.
+      #
+      # @example
+      #   # bad
+      #   <<-SQL
+      #     SELECT * FROM posts;
+      #   SQL
+      #
+      #   <<-SQL
+      #     SELECT * FROM posts
+      #       WHERE id = 1
+      #   SQL
+      #
+      #   execute(<<~SQL, "Post Load")
+      #     SELECT * FROM posts
+      #       WHERE post_id = 1
+      #   SQL
+      #
+      #   # good
+      #   <<-SQL.squish
+      #     SELECT * FROM posts;
+      #   SQL
+      #
+      #   <<~SQL.squish
+      #     SELECT * FROM table
+      #       WHERE id = 1
+      #   SQL
+      #
+      #   execute(<<~SQL.squish, "Post Load")
+      #     SELECT * FROM posts
+      #       WHERE post_id = 1
+      #   SQL
+      #
+      class SquishedSQLHeredocs < Cop
+        include Heredoc
+        include RangeHelp
+
+        SQL = 'SQL'
+        SQUISH = '.squish'
+        SQL_SQUISH = "#{SQL}#{SQUISH}"
+        MSG = 'Use `%<expect>s` instead of `%<current>s`.'
+
+        def on_heredoc(node)
+          return unless offense_detected?(node)
+
+          add_offense(node)
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.insert_after(node, SQUISH)
+          end
+        end
+
+        private
+
+        def offense_detected?(node)
+          sql_heredoc?(node) && offense?(node)
+        end
+
+        def sql_heredoc?(node)
+          delimiter_string(node) == SQL
+        end
+
+        def offense?(node)
+          !using_squish_on_sql_heredoc?(node)
+        end
+
+        def using_squish_on_sql_heredoc?(node)
+          node.parent&.send_type? && node.parent&.method?(:squish)
+        end
+
+        def message(node)
+          format(
+            MSG,
+            expect: "#{heredoc_type(node)}#{SQL_SQUISH}",
```

Comment:
This seems simpler as:

```suggestion
            expect: "#{node.source}#{SQUISH}",
```

That would also make `SQL_SQUISH` unnecessary.